vendor/k8s.io/kubernetes/pkg/kubelet/kubelet_server_journal.go

```diff
@@ -135,6 +135,16 @@ func (a *journalArgs) Args() []string {
 // args to the provided writer, timing out at a.Timeout. If an error occurs a line
 // is written to the output.
 func (a *journalArgs) Copy(w io.Writer) {
+	// determine whether we have more than a single boot record, and if so, include the previous
+	// boot so that we always have logs from before a restart
+	args := a.Args()
+	args = append(args, "--boot")
+	if out, err := exec.Command("journalctl", "--list-boot").CombinedOutput(); err == nil && bytes.Count(bytes.TrimSpace(out), []byte("\n")) > 1 {
```

Comment:
Do we need to preflight this with `--list-boots`?  Why not just try?  `journalctl` seems to handle non-existent IDs gracefully:

```console
$ sudo journalctl --boot -1000
Failed to look up boot -1000: No such boot ID in journal
$ echo $?
1
```